architecture/transaction-layer.md

```diff
@@ -0,0 +1,204 @@
+
+For more detail about the risks that large clock offsets can cause, see [What happens when node clocks are not properly synchronized?](operational-faqs.html#what-happens-when-node-clocks-are-not-properly-synchronized)
+
+#### Timestamps & Serializability
+
+HLC timestamps let us understand the sequence of state changes in the database. CockroachDB then uses this ordering to provide its isolation guarantees. Isolation, in this context, meaning the order in which keys and their values become accessible to other operations. By providing strict isolation levels, we aim to provide anomaly-free consistency to our users, which we refer to as "serializability"––the database's state is ordered in a strictly serialized way, clearly ordered by HLC timestamps.
+
+For instance, if an operation finds a value with a timestamp in the future, it "understands" that its own timestamp must be moved forward to preserve serialized order in the database.
+
+However, when timestamps are moved, it introduces anomalies into the database. This is an intuitive concept, that it violates the notion of strict serialiazability to reorder the state of your data's values.
+
+In this document, you can find more about our [isolation levels]().
+
+### Timestamp Cache
+
+To provide provide serializability, whenever an operation reads a value, we store the operation's timestamp in a Timestamp Cache, which shows the high-water mark for values being read.
+
+Whenever writes occur, their timestamps are checked against the TimestampCache to ensure its timestamp is greater than the Timestamp Cache (otherwise we would allow you to rewrite history, which produces anomalies). If the timestamp is less than, we attempt to move the timestamp to a time later timestamp than the cache.
+
+### client.Txn and TxnCoordSender
+
+As we mentioned in the SQL layer's architectural overview, CockroachDB converts all SQL statements into key-value (KV) operations, which is how data is ultimately stored and accessed. 
+
+All of the KV operations generated from the SQL layer's use `client.Txn`, which is the transactional interface for the CockroachDB KV layer––but, as we discussed above, all statements are treated as transactions, so all statements use this interface.
+
+However, `client.Txn` is actually just a wrapper around `TxnCoordSender`, which plays a crucial role in our code base by:
+
+- Dealing with transactions' state. After a transaction is started, `TxnCoordSender` starts asynchronously sending heartbeat messages to that transaction's Transaction Record, which signal that it should be kept alive. If the `TxnCoordSender`'s heartbeating stops, the Transaction Record is moved to `ABORTED` status.
```

Comment:
`which signal` > `which signals`

`move to ABORTED status` > `moved to the ABORTED status`